motoman_resources/urdf/common_macros.xacro

```diff
@@ -0,0 +1,234 @@
+<?xml version="1.0"?>
```

Comment:
I would recommend not using this macro here.

There are other ways to load `.yaml` data which are nicer, more maintainable and easier to understand.

Reply:
Fair enough, any suggestions?

I was just using this because in the back of my mind I am considering trying to do something akin to what the UR descriptions package has going on, with one file for all the arms then config files to make all the different robots. This seemed like a decent way to consistently load up the values, dictating names in the configs with the xacro soruce.
Obviously it won't work for all the Yaskawa arms, but it seems like many of them fall into the standard 6 dof arm category.

Reply:
If you don't think there's any value in a future shift like that, I can abandon this altogether. Not hard set on the idea, was just a thought.

Reply:
It's a nice idea, but I'm not sold it on (yet).

There are a couple of reasons we duplicate so much in these repositories:

 - it's easier to grok what is going on (I know I helped rewrite the `universal_robot` xacros to rely on `.yaml` for just about everything, but tbh it doesn't really get any clearer)
 - easier to make changes to individual models (ie: things which are not part of the `.yaml`)
 - less moving parts
 - packages have less dependencies
 - not all xacro parsers actually support loading `.yaml` files (several are in use in external projects, which will not be able to load the models any more)

Especially the first bullet is not something to underestimate. Cognitive overload is easy to achieve, and with everything going in ROS, adding more complexity isn't always a good thing.

As I wrote: I'm not against it per se, I just think we should be a bit careful about importing the way it was done in `universal_robot`.

An alternative could be to pass entire dicts from the `.yaml` to `xacro:macro`s, as Robert describes here: https://github.com/ros/xacro/issues/240#issuecomment-636302854.

Reply:
Alright, for my purposes I only _need_ the kinematics as input at the moment.

What do you think of leaving the nominal kinematic values in the URDF, and load offsets from nominal from a yaml file for the purpose of calibration? This wouldn't involve a separate macro, the dictionary containing the values would be the xacro argument.

Doesn't have the complexity and flexibility of the other approach, still allows for calibration, but makes the nominal values clear for users that do not care about that, and the loading of the yaml optional (can fallback to all 0s for offsets if no dictionary is passed in)